src/cogent3/core/annotation_db.py

```diff
@@ -1622,3 +1631,77 @@ def load_annotations(
             show_progress=show_progress,
         )
     )
+
+
+def _update_array_format(data: bytes) -> bytes:
+    """Convert from the old to the new sqlite numpy array format.
+
+    Converts the previous format saved with numpy.ndarray.tobytes
+    to the .npy format generated with numpy.save.
+
+    Ignores any entries saved in the new format
+
+    Parameters
+    ----------
+    data : bytes
+        The sqlite3 representation of the numpy array.
+
+    Returns
+    -------
+    bytes
+        The new sqlite3 representation of the numpy array.
+    """
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        new_data = array_to_sqlite(sqlite_to_array(data))
+    return new_data
+
+
+def update_file_format(
+    source_file: os.PathLike,
+    backup_file: typing.Optional[os.PathLike],
```

Comment:
Another way to do this is make backup "opt-out" by the user, so make `backup_file=True` the default. We can just append `.bak` to the original file path.

Reply:
Good idea!